contracts/pcv/balancer/BalancerPCVDepositBase.sol

```diff
@@ -0,0 +1,169 @@
+        uint256 _maximumSlippageBasisPoints
+    ) CoreRef(_core) {
+        vault = IVault(_vault);
+        rewards = IMerkleOrchard(_rewards);
+        maximumSlippageBasisPoints = _maximumSlippageBasisPoints;
+        poolId = _poolId;
+
+        (poolAddress, ) = IVault(_vault).getPool(_poolId);
+
+        // get the balancer pool tokens
+        IERC20[] memory tokens;
+        (tokens, , ) = IVault(_vault).getPoolTokens(_poolId);
+
+        // cache the balancer pool tokens as Assets
+        poolAssets = new IAsset[](tokens.length);
+        for (uint256 i = 0; i < tokens.length; i++) {
+            poolAssets[i] = IAsset(address(tokens[i]));
+        }
+    }
+
+    // Accept ETH transfers
+    receive() external payable {}
+
+    /// @notice Wraps all ETH held by the contract to WETH
+    /// Anyone can call it.
+    /// Balancer uses WETH in its pools, and not ETH.
+    function wrapETH() external {
+        uint256 ethBalance = address(this).balance;
+        if (ethBalance != 0) {
+            Constants.WETH.deposit{value: ethBalance}();
+        }
+    }
+
+    /// @notice unwrap WETH on the contract, for instance before
+    /// sending to another PCVDeposit that needs pure ETH.
+    /// Balancer uses WETH in its pools, and not ETH.
+    function unwrapETH() external onlyPCVController {
+        uint256 wethBalance = IERC20(address(Constants.WETH)).balanceOf(address(this));
+        if (wethBalance != 0) {
+            Constants.WETH.withdraw(wethBalance);
+        }
+    }
+
+    /// @notice Sets the maximum slippage vs 1:1 price accepted during withdraw.
+    /// @param _maximumSlippageBasisPoints the maximum slippage expressed in basis points (1/10_000)
+    function setMaximumSlippage(uint256 _maximumSlippageBasisPoints) external onlyGovernorOrAdmin {
+        require(_maximumSlippageBasisPoints <= Constants.BASIS_POINTS_GRANULARITY, "EthLidoPCVDeposit: Exceeds bp granularity.");
+        maximumSlippageBasisPoints = _maximumSlippageBasisPoints;
+        emit UpdateMaximumSlippage(_maximumSlippageBasisPoints);
+    }
+
+    /// @notice redeeem all assets from LP pool
+    /// @param to destination for withdrawn tokens
+    function exitPool(address to) external whenNotPaused onlyPCVController {
+        uint256 bptBalance = IWeightedPool(poolAddress).balanceOf(address(this));
+        if (bptBalance != 0) {
+            IVault.ExitPoolRequest memory request;
+
+            // Uses encoding for exact BPT IN withdrawal using all held BPT
+            bytes memory userData = abi.encode(IWeightedPool.ExitKind.EXACT_BPT_IN_FOR_TOKENS_OUT, bptBalance);
+            request.assets = poolAssets;
+            request.minAmountsOut = new uint256[](2); // 0 min
+            request.userData = userData;
+            request.toInternalBalance = false; // use external balances to be able to transfer out tokenReceived
+
+            vault.exitPool(poolId, address(this), payable(to), request);
+
+            _burnFeiHeld();
```

Comment:
is this necessary given the `to` address could be somewhere else?

Reply:
I removed the `to`, would be a mess if FEI was sent to other contracts that can't burn it etc. I think it's more clever to just have one `exitPool()` that redeem underlying on the contract, and then use `withdrawERC20` to move underlying assets as needed